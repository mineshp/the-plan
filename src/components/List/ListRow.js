import React from 'react';
import PropTypes from 'prop-types';
import { Button, Table, Icon, Label } from 'semantic-ui-react';

const projectLabels = (projects) => projects.map((project) => (
    <Label
        as="a"
        key={project.id}
        color="red"
    >
        <Icon name="empty star" />
        {project.name}
    </Label>
));

const ListRow = ({ data }) => {
    const listID = data._id; // eslint-disable-line no-underscore-dangle
    return (
        <Table.Row>
            <Table.Cell>{data.listName}</Table.Cell>
            <Table.Cell>{data.createdDate}</Table.Cell>
            <Table.Cell>{data.updatedDate}</Table.Cell>
            <Table.Cell>{projectLabels(data.projects)}</Table.Cell>
            <Table.Cell><Button as="a" href={`/list/${listID}/view`}>View</Button></Table.Cell>
        </Table.Row>
    );
};

ListRow.propTypes = {
    data: PropTypes.shape({
        _id: PropTypes.string.isRequired,
        listName: PropTypes.string.isRequired
    }).isRequired
};

export default ListRow;
