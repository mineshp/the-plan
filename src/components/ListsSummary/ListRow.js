import React from 'react';
import PropTypes from 'prop-types';
import { Button, Table, Icon, Label } from 'semantic-ui-react';
import { formatDate } from '../../helpers/validators/common';

const projectLabels = (projects) => projects.map((project) => (
    <Label
        as="a"
        key={project.id}
        color="brown"
    >
        <Icon name="empty star" />
        {project.name}
    </Label>
));

const List = ({ data, onDeleteHandler }) => {
    const listID = data._id; // eslint-disable-line no-underscore-dangle
    return (
        <Table.Row>
            <Table.Cell>{data.listName}</Table.Cell>
            <Table.Cell>{formatDate(data.createdDate)}</Table.Cell>
            <Table.Cell>{formatDate(data.updatedDate)}</Table.Cell>
            <Table.Cell>{projectLabels(data.projects)}</Table.Cell>
            <Table.Cell>
                <Button
                    as="a"
                    color="blue"
                    icon="list layout"
                    href={`/list/view/${listID}`}
                />
                <Button
                    as="a"
                    icon="edit"
                    color="green"
                    href={`/list/update/${listID}`}
                />
                <Button
                    icon="trash"
                    color="pink"
                    value={listID}
                    onClick={onDeleteHandler}
                />
            </Table.Cell>
        </Table.Row>
    );
};

List.propTypes = {
    data: PropTypes.shape({
        _id: PropTypes.string.isRequired,
        listName: PropTypes.string.isRequired
    }).isRequired,
    onDeleteHandler: PropTypes.func.isRequired
};

export default List;
