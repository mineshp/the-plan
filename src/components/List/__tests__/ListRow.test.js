import React from 'react';
import ListRow from '../ListRow';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Label } from 'semantic-ui-react';

const mockListRow = {
    _id: '1234456',
    listName: 'My Awesome List',
    projects: [{
        id: '1',
        name: 'ABC'
    },
    {
        id: '2',
        name: 'XYZ'
    }]
};

describe('ListRow', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <ListRow key={mockListRow._id} data={mockListRow} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the correct number of project labels', () => {
        const wrapper = shallow(<ListRow key={mockListRow._id} data={mockListRow} />);
        const ProjectLabels = wrapper.find(Label);

        expect(ProjectLabels.length).toBe(2);
    });
});