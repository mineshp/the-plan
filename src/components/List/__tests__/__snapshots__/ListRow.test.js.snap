// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListRow renders correctly 1`] = `
<tr
  className=""
>
  <td
    className=""
  >
    My Awesome List
  </td>
  <td
    className=""
  />
  <td
    className=""
  />
  <td
    className=""
  >
    <a
      className="ui red label"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="empty star icon"
      />
      ABC
    </a>
    <a
      className="ui red label"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="empty star icon"
      />
      XYZ
    </a>
  </td>
  <td
    className=""
  >
    <a
      className="ui button"
      disabled={undefined}
      href="/list/1234456/view"
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      View
    </a>
  </td>
</tr>
`;
